preload/prefetch-events.html

```diff
@@ -0,0 +1,72 @@
+<!DOCTYPE html>
+<title>Ensures that prefetch respects HTTP cache semantics</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/utils.js"></script>
+<script src="resources/prefetch-helper.js"></script>
+<script src="/common/get-host-info.sub.js"></script>
+<body>
+<script>
+const {REMOTE_ORIGIN} = get_host_info();
+async function prefetch(link, uid) {
+    link.rel = "prefetch";
+    document.head.appendChild(link);
+    const event = new Promise(resolve => {
+        link.addEventListener("error", () => resolve("error"));
+        link.addEventListener("load", () => resolve("load"));
+        setTimeout(() => resolve("timeout"), 1000);
+    });
+    return await event;
+}
+
+promise_test(async t => {
+    const uid = token();
+    const link = document.createElement("link");
+    link.href = `/preload/resources/prefetch-info.py?key=${uid}`;
+    const event = await prefetch(link, uid);
+    assert_equals(event, "load");
+}, "Prefetch should fire the load event");
+
+promise_test(async t => {
+    const uid = token();
+    const link = document.createElement("link");
+    link.href = `/preload/resources/prefetch-info.py?key=${uid}&status=404`;
+    const event = await prefetch(link, uid);
+    assert_equals(event, "error");
+}, "Prefetch should fire the error event for 404");
+
+promise_test(async t => {
+    const uid = token();
+    const link = document.createElement("link");
+    link.href = `/preload/resources/prefetch-info.py?key=${uid}&status=500`;
+    const event = await prefetch(link, uid);
+    assert_equals(event, "error");
+}, "Prefetch should fire the error event for 500");
```

Comment:
Can we test 404s and/or 500s for remote origins? It's a bit surprising we'd leak that information across origins, although I think it happens with other elements too so if so that's fine.

Reply:
Done